Fix the shadows that style the code example line numbers

r0ckarong opened the pull request:
I was playing around with the color theme for the code examples and noticed that after 626 lines the formatting of the line number broke. You use some box shadows that extend 9999 pixels in two directions to style that area. Unfortunately with long code examples this is not enough since half a screen page on a FHD monitor is already around 500pixels. 

Also the second shadow that styles the "border" fencing off the line number area from the actual code area was a hardcoded color and wouldn't follow darkmode or any other color settings. I just swapped the hardcoded rgba value for the color of the actual line number, that gives enough contrast and looks nice.

![shadow_error](https://user-images.githubusercontent.com/4920542/117444932-66a53880-af3a-11eb-8266-72ec486a4574.png)

After extending the shadow area by a cool 990000 pixels to avoid future problems:
![shadow_error_million](https://user-images.githubusercontent.com/4920542/117444931-660ca200-af3a-11eb-80d8-0cbe7eb01fab.png)

jothepro commented:
That's the moment when the dirty hacks are starting to haunt you 😆 

Apparently the default code fragment theme uses different colors for line numbers & the border. I tested your change with it and think that a white border line does look a bit too harsh in the default theme:
<img width="690" alt="Bildschirmfoto 2021-05-07 um 14 05 27" src="https://user-images.githubusercontent.com/21294002/117447062-488d0780-af3d-11eb-9231-a6903d5fe0f1.png">

I agree that the border color should be configurable, but it has to be a separate CSS variable in order to work well with the default theme.

r0ckarong commented:
> 
> 
> That's the moment when the dirty hacks are starting to haunt me 😆
> 
> Apparently the default code fragment theme uses different colors for line numbers & the border. I tested your change with it and think that a white border line does look a bit too harsh in the default theme:
> <img alt="Bildschirmfoto 2021-05-07 um 14 05 27" width="690" src="https://user-images.githubusercontent.com/21294002/117447062-488d0780-af3d-11eb-9231-a6903d5fe0f1.png">
> 
> I agree that the border color should be configurable, but it has to be a separate CSS variable in order to work well with the default theme.

Then I would opt for a `--fragment-linenumber-border` variable that can be set. In my case I would just fill that with `var(--fragmen-linenumber-color)` but then everybody can decide how they want to solve it. I can amend the PR with this change if you want.

r0ckarong commented:
I've now added a very subtle grey border color and changed the shorthand codes to long ones. They looked out of place in the list.

![shadow_basetheme](https://user-images.githubusercontent.com/4920542/117449089-d9fd7900-af3f-11eb-868b-0fd1036771d3.png)

jothepro commented on doxygen-awesome.css:
```diff
@@ -131,16 +131,17 @@ SOFTWARE.
-    --fragment-linenumber-color: #ccc;
+    --fragment-linenumber-color: #cccccc;
     --fragment-linenumber-background: #35393c;
+    --fragment-linenumber-border: #454545;
```
```suggestion
    --fragment-linenumber-border: #1f1f1f;
```

this is closer to the current default theme border color.

r0ckarong replied:
Yeah that works better for the default theme.

jothepro commented:
Thank you for your contribution, @r0ckarong !
